x-pack/plugins/cloud_integrations/cloud_experiments/server/plugin.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+} from '@kbn/core/server';
+import { get, has } from 'lodash';
+import LaunchDarkly, { type LDClient, type LDUser } from 'launchdarkly-node-server-sdk';
+import type { LogMeta } from '@kbn/logging';
+import type { UsageCollectionSetup } from '@kbn/usage-collection-plugin/server';
+import { registerUsageCollector } from './usage';
+import type { CloudExperimentsConfigType } from './config';
+import type {
+  CloudExperimentsFeatureFlagNames,
+  CloudExperimentsMetric,
+  CloudExperimentsPluginSetup,
+  CloudExperimentsPluginStart,
+} from '../common';
+import { FEATURE_FLAG_NAMES, METRIC_NAMES } from '../common/constants';
+
+interface CloudExperimentsPluginSetupDeps {
+  usageCollection?: UsageCollectionSetup;
+}
+
+export class CloudExperimentsPlugin
+  implements
+    Plugin<
+      CloudExperimentsPluginSetup,
+      CloudExperimentsPluginStart,
+      CloudExperimentsPluginSetupDeps
+    >
+{
+  private readonly logger: Logger;
+  private readonly launchDarklyClient?: LDClient;
+  private readonly flagOverrides?: Record<string, unknown>;
+  private launchDarklyUser: LDUser | undefined;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+    const config = initializerContext.config.get<CloudExperimentsConfigType>();
+    if (config.flag_overrides) {
+      this.flagOverrides = config.flag_overrides;
+    }
+    const ldConfig = config.launch_darkly; // If the plugin is enabled and no flag_overrides are provided (dev mode only), launch_darkly must exist
+    if (!ldConfig && !initializerContext.env.mode.dev) {
+      // If the plugin is enabled, and it's in prod mode, launch_darkly must exist
+      // (config-schema should enforce it, but just in case).
+      throw new Error(
+        'xpack.cloud_integrations.experiments.launch_darkly configuration should exist'
+      );
+    }
+    if (ldConfig) {
+      this.launchDarklyClient = LaunchDarkly.init(ldConfig.sdk_key, {
```

Comment:
optional idea to consider for the future: Organizationally, it might be nice to have all of the LD-specific logic pulled into a single class, so we have a thin wrapper for the methods that we actually need to use and simply call the wrapper from our `plugin.ts` after passing in any dependencies. But that's a super nit and really more about readability / testability than anything else, so it's not worth blocking this PR on it.

Reply:
I see where you're coming from. The way I was looking at it is that the plugin is the wrapper. But I see value in your comment.

I'm happy to extract this to a wrapper.

Reply:
I've been playing around, and it requires a good amount of refactoring (to your point 😅 ), especially in the tests. 

If that's fine, I'll submit those changes in a separate PR.

Reply:
Definitely not a blocker. I agree, for now the plugin class more or less functions as the wrapper, so I don't see this as an urgent thing, just a nit